### PR TITLE
Remove agents

### PR DESCRIPTION
**Pull request by mattmundell**
**Checklist**:

- [ ] Tests
- [x] [CHANGELOG](https://github.com/greenbone/gvmd/blob/master/CHANGELOG.md) Entry

### CODE REVIEW

**Comment by mattmundell** [comment]
Not sure what happened with the GMP doc diff.  Let me know if it's too hard to follow and I'll break it up.

File: src/manage_migrators.c [hunk 1]
```diff
@@ -1633,6 +1633,77 @@ migrate_222_to_223 ()
...
+    {
+      sql_rollback ();
+      return -1;
+    }
+
+  /* Update the database. */
+
+  /* Agents were removed entirely. */
+
+  sql ("DELETE FROM tags"
```
Comment (timopollmeier) [inline src/manage_migrators.c]: Deleting tags that are still referenced by `tag_resources` will violate the foreign key constraint. 

The simplest solution would be to just remove this `DELETE` because it would only cover tags that have no `resource_type` themselves or wrong ones.

If you still want to delete all the tags referencing agents, you would have to collect them before removing the `tag_resources` rows and then delete them later. 
Also, I think the condition should be ` WHERE id IN (SELECT ...)` instead of ` WHERE id = (SELECT ...)`.

Comment (mattmundell) [reply src/manage_migrators.c]: I removed the DELETE in https://github.com/greenbone/gvmd/pull/922/commits/904e6ed17f253966c0f87a3e2d9cbc6a1abaf9c2.

And yes, you're right about the IN.
